api/controller.go

```diff
@@ -2801,64 +2791,50 @@ func (c *Controller) ActionsListRunsHandler() actionsop.ListRunsHandler {
 				WithPayload(responseErrorFrom(err))
 		}
 
-		before := time.Now()
-		if params.Before != nil {
-			before = time.Time(*params.Before)
-		}
-		runsIter, err := deps.Actions.ListRuns(params.Repository, before, params.Branch)
+		after := swag.StringValue(params.After)
+		runsIter, err := deps.Actions.ListRuns(deps.ctx, params.Repository, params.Branch, after)
 		if err != nil {
 			if errors.Is(err, actions.ErrNotFound) {
 				return actionsop.NewListRunsNotFound().
 					WithPayload(responseErrorFrom(err))
 			}
-
 			return actionsop.NewListRunsDefault(http.StatusInternalServerError).
 				WithPayload(responseErrorFrom(err))
 		}
 
-		res := &actionsop.ListRunsOKBody{Pagination: &models.Pagination{HasMore: swag.Bool(true)}}
-		for runsIter.Next() {
+		payload := &actionsop.ListRunsOKBody{
+			Results: make([]*models.ActionRun, 0),
```

Comment:
```suggestion
			Results: make([]*models.ActionRun),
```

Reply:
This is on purpose so the caller doesn't need to handle nil - will get empty array